src/hb-aat-layout-trak-table.hh

```diff
@@ -82,9 +120,38 @@ struct trak
   inline bool sanitize (hb_sanitize_context_t *c) const
   {
     TRACE_SANITIZE (this);
+    /* TODO */
     return_trace (c->check_struct (this));
   }
 
+  inline bool apply (hb_aat_apply_context_t *c) const
+  {
+    TRACE_APPLY (this);
+    const float ptem = c->font->ptem;
+    if (ptem > 0.f)
+    {
+      hb_buffer_t *buffer = c->buffer;
+      if (HB_DIRECTION_IS_HORIZONTAL (c->buffer->props.direction))
+      {
+        const TrackData trackData = this+horizOffset;
+        float tracking = trackData.get_tracking (this, ptem);
+        int x_scale = c->font->x_scale / c->face->upem;
+        // TODO: Do we anything else for RTL?
+        foreach_grapheme (buffer, start, end)
+          buffer->pos[end].x_advance += tracking * x_scale;
+      }
+      else
+      {
+        const TrackData trackData = this+vertOffset;
+        float tracking = trackData.get_tracking (this, ptem);
+        int y_scale = c->font->y_scale / c->face->upem;
+        foreach_grapheme (buffer, start, end)
+          buffer->pos[end].y_advance += tracking * y_scale;
```

Comment:
Anything to be worried about TTB/BTT distinction?

Reply:
In general I think we want to add half of the tracking to each edge of the glyph...

Reply:
Done